stable/artifactory-ha/templates/nginx-deployment.yaml

```diff
@@ -118,7 +124,13 @@ spec:
         livenessProbe:
           httpGet:
             path: {{ .Values.nginx.livenessProbe.path }}
-            port: 80
+            {{- if .Values.nginx.http.enabled }}
+            port: {{ .Values.nginx.http.internalPort }}
+            scheme: HTTP
+            {{- else }}
+            port: {{ .Values.nginx.https.internalPort }}
+            scheme: HTTPS
```

Comment:
what about SSL certs? This can be an issue when you try to use https probes with ip addresses (which is what k8s does from what I know)

Reply:
Good question. It seems to be working on my current set up with the following:

Values.yaml
```yaml
artifactory:
  deleteDBPropertiesOnStartup: true
  node:
    javaOpts:
      jmx:
        enabled: false
      xms: 6g
      xmx: 8g
    nodeSelector:
      jfrog-role: member
    resources:
      limits:
        cpu: 7
        memory: 12Gi
      requests:
        cpu: 4
        memory: 6Gi
  primary:
    javaOpts:
      jmx:
        enabled: false
      xms: 3g
      xmx: 4g
    nodeSelector:
      jfrog-role: primary
    resources:
      limits:
        cpu: 3
        memory: 7Gi
      requests:
        cpu: 2
        memory: 3Gi
databaseUpgradeReady: true
nginx:
  http:
    enabled: false
  https:
    externalPort: 443
    internalPort: 443
  resources:
    limits:
      cpu: 250m
      memory: 500Mi
    requests:
      cpu: 100m
      memory: 250Mi
  service:
    annotations:
      service.beta.kubernetes.io/aws-load-balancer-backend-protocol: https
      service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0
      service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn
      service.beta.kubernetes.io/aws-load-balancer-ssl-ports: https
```

nginx-deployment snippet:
```yaml
spec:
      containers:
      - command:
        - nginx
        - -g
        - daemon off;
        image: docker.bintray.io/jfrog/nginx-artifactory-pro:6.16.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          failureThreshold: 10
          httpGet:
            path: /artifactory/webapp/#/login
            port: 443
            scheme: HTTPS
          initialDelaySeconds: 60
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 10
        name: nginx
        ports:
        - containerPort: 443
          protocol: TCP
        readinessProbe:
          failureThreshold: 10
          httpGet:
            path: /artifactory/webapp/#/login
            port: 443
            scheme: HTTPS
          initialDelaySeconds: 60
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 10
        resources:
          limits:
            cpu: 250m
            memory: 500Mi
          requests:
            cpu: 100m
            memory: 250Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /etc/nginx/nginx.conf
          name: nginx-conf
          subPath: nginx.conf
        - mountPath: /var/opt/jfrog/nginx/conf.d/
          name: nginx-artifactory-conf
        - mountPath: /var/opt/jfrog/nginx
          name: nginx-volume
        - mountPath: /var/opt/jfrog/nginx/ssl
          name: ssl-certificates
      dnsPolicy: ClusterFirst
      initContainers:
      - command:
        - /bin/sh
        - -c
        - |
          rm -rfv /var/opt/jfrog/nginx/lost+found; mkdir -p /var/opt/jfrog/nginx/logs;
        image: alpine:3.10
        imagePullPolicy: IfNotPresent
        name: setup
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /var/opt/jfrog/nginx
          name: nginx-volume
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext:
        fsGroup: 107
        runAsUser: 104
      serviceAccount: artifactory-ha
      serviceAccountName: artifactory-ha
      terminationGracePeriodSeconds: 30
      volumes:
      - configMap:
          defaultMode: 420
          name: artifactory-ha-nginx-conf
        name: nginx-conf
      - configMap:
          defaultMode: 420
          name: artifactory-ha-nginx-artifactory-conf
        name: nginx-artifactory-conf
      - emptyDir: {}
        name: nginx-volume
      - name: ssl-certificates
        secret:
          defaultMode: 420
          secretName: artifactory-ha-nginx-certificate
```

Reply:
mm ok cool. LGTM